### ISSUE TITLE
Parallelization Bug: Unable to find built xctestrun file

### ISSUE THREAD

**Issue by lyndsey-ferguson** [lyndsey-ferguson/fastlane-plugin-test_center#105]
### New Issue Checklist

- [x] Updated `fastlane-plugin-test_center` to the latest version
- [x] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [x] I reviewed the [example(s)](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md) for the action(s) I am using
- [x] I have removed any sensitive data such as passwords, authentication tokens, or anything else I do not want to world to see

### Issue Description

This is using the parallelized `multi_scan` described [here](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/pull/104). It crashes when trying to find the xctestrun file.

The code to find the xctestrun file is making some assumptions about how it is named. Instead, the code should be examining the project file to get the exact name of the file.

##### Complete output when running fastlane, including the stack trace and command used

<details>
:)

Console log:
```shell
+--------------------+---+
|      Test Results      |
+--------------------+---+
| Number of tests    | 0 |
| Number of failures | 0 |
+--------------------+---+

[08:45:49]: Error: cannot find xctestrun file ''
[08:45:49]: --------------------------
[08:45:49]: --- Step: xchtmlreport ---
[08:45:49]: --------------------------
+------------------+--------------------------+
|                Lane Context                 |
+------------------+--------------------------+
| DEFAULT_PLATFORM | ios                      |
| PLATFORM_NAME    | ios                      |
| LANE_NAME        | ios run_automation_tests |
+------------------+--------------------------+
[08:45:49]: You must pass at least one result_bundle_path

+------+------------------------------------+-------------+
|                    fastlane summary                     |
+------+------------------------------------+-------------+
| Step | Action                             | Time (in s) |
+------+------------------------------------+-------------+
| 1    | Verifying fastlane version         | 0           |
| 2    | opt_out_usage                      | 0           |
| 3    | default_platform                   | 0           |
| 4    | Switch to ios setup_simulator lane | 0           |
| 5    | set_scan_xcargs                    | 0           |
| 💥   | multi_scan                         | 15          |
| 💥   | xchtmlreport                       | 0           |
+------+------------------------------------+-------------+

[08:45:49]: fastlane finished with errors

[!] You must pass at least one result_bundle_path

```
Fastlane looks like this

```ruby
begin
      test_run_block = lambda do |testrun_info| 
        failed_test_count = testrun_info[:failed].size
        
        if failed_test_count > 0
          UI.important('The run of tests would finish with failures due to fragile tests here.')
    
          try_attempt = testrun_info[:try_count]
          if try_attempt < 3
            UI.header('Since we are using :multi_scan, we can re-run just those failing tests!')
          end
        end
      end
      
      result = multi_scan(
        workspace: File.absolute_path('../Com.xcworkspace'),
        scheme: 'Automation',
        try_count: 3,
        parallel_testrun_count: 4,
        fail_build: false,
        # testrun_completed_block: test_run_block,
        # output_files: 'report.html',
        # output_types: 'html'
      )
      unless result[:failed_testcount].zero?
        UI.message("There are #{result[:failed_testcount]} legitimate failing tests")
      end
    rescue => ex
      # Logs error instead of failing build
      UI.error(ex)
    ensure
      xchtmlreport(enable_junit: true)
    end
  end
```
</details>

**Comment by lyndsey-ferguson** [comment]
@ishammoho  please let me know what you find by following my request [here](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/issues/10#issuecomment-504206152)

**Comment by lyndsey-ferguson** [comment]
From @ishammoho

Awesome. Thanks @lyndsey-ferguson 

passed this to multi_scan
xctestrun: "./DerivedData/Build/Products/Automation_iphonesimulator12.1-x86_64.xctestrun",

and it's working now

However, finishes with You must pass at least one result_bundle_path error which I mentioned earlier.

The default multi_scan reporting is not showing all the tests which have run. 

```
[13:22:25]: --------------------------
[13:22:25]: --- Step: xchtmlreport ---
[13:22:25]: --------------------------
+------------------+--------------------------+
|                Lane Context                 |
+------------------+--------------------------+
| DEFAULT_PLATFORM | ios                      |
| PLATFORM_NAME    | ios                      |
| LANE_NAME        | ios run_automation_tests |
+------------------+--------------------------+
[13:22:25]: You must pass at least one result_bundle_path

+------+------------------------------------+-------------+
|                    fastlane summary                     |
+------+------------------------------------+-------------+
| Step | Action                             | Time (in s) |
+------+------------------------------------+-------------+
| 1    | Verifying fastlane version         | 0           |
| 2    | opt_out_usage                      | 0           |
| 3    | default_platform                   | 0           |
| 4    | Switch to ios setup_simulator lane | 0           |
| 5    | set_scan_xcargs                    | 0           |
| 6    | multi_scan                         | 1493        |
| 💥   | xchtmlreport                       | 0           |
+------+------------------------------------+-------------+

[13:22:25]: fastlane finished with errors

[!] You must pass at least one result_bundle_path

```

**Comment by lyndsey-ferguson** [comment]
Also from @ishammoho

<img width="295" alt="Screen Shot 2019-06-21 at 2 32 57 PM" src="https://user-images.githubusercontent.com/6156112/59893668-c4f1cc80-9431-11e9-8548-3e4f39c9c62e.png">
When test starts it generates 4 different folders (i have asked to run 4 parallel tests). Which has the result_bundle as well. However when the test finished it cleans all 4 of these bundles (folders). I'm assuming if you don't clean it up, reporting could be generated using that?

**Comment by lyndsey-ferguson** [comment]
@ishammoho

> However, finishes with You must pass at least one result_bundle_path error which I mentioned earlier.
Correct, `xchtmlreport` is failing because it is not receiving the `test_result` bundle path.

> However when the test finished it cleans all 4 of these bundles (folders). I'm assuming if you don't clean it up, reporting could be generated using that?

Yes, I believe you could use that. I'll create a separate Issue for that. In the meantime, you can run `xchtmlreport` on each `test_result` in your `test_run_block`. You can get the dir path of one of those batch folders from `testrun_info[:report_filepath]` using `File.dirname` on that path.

**Comment by lyndsey-ferguson** [comment]
@ishammoho please pass `result_bundle` to `multi_scan` to place a copy of the `test_result` bundle in your output directory. That should also handle collating such reports.

**Comment by lyndsey-ferguson** [comment]
@ishammoho I've created [Parallelization Bug: generated `test_result` bundles are deleted after completion](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/issues/106) to track progress on the other issue you found.

**Comment by lyndsey-ferguson** [comment]
@ishammoho can you modify your `Pluginfile` per my instructions below, remove your `Gemfile.lock`, run `bundle install`, modify your `Fastfile` to *no longer* pass in the `xctestrun` option, and then run your fastlane again?

Pluginfile:
```
gem 'fastlane-plugin-test_center', :git => "https://github.com/lyndsey-ferguson/fastlane-plugin-test_center.git", :branch => "issue-10-parellel-multi_scan"
```

Please attach the console output as a text file to this issue if it still is a problem (makes it easier for me to review).

**Comment by lyndsey-ferguson** [comment]
Fixed in [v3.8.0.parallelizing.beta.3](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/releases/tag/v3.8.0.parallelizing.beta.3)

**Comment by ishammoho** [comment]
Thanks @lyndsey-ferguson . It's working, not need to pass xctestrun option anymore. Thanks. Trying to sort the reporting part as you suggested. Will update you on that.
